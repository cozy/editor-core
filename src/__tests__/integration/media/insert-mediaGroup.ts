import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { editable, getDocFromElement, comment, insertMedia } from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';

[comment].forEach(editor => {
  BrowserTestCase(
    `insert-mediaGroup.ts: Inserts a media group on ${editor.name}`,
    { skip: ['edge', 'safari'] },
    async (
      client: Parameters<typeof goToEditorTestingExample>[0],
      testName: string,
    ) => {
      const page = await goToEditorTestingExample(client);
      await mountEditor(page, {
        appearance: editor.appearance,
        media: {
          allowMediaSingle: false,
          allowMediaGroup: true,
        },
      });

      await page.click(editable);
      await page.type(editable, 'some text');

      // now we can insert media as necessary
      await insertMedia(page);

      expect(await page.isVisible('.wrapper')).toBe(true);

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );

  BrowserTestCase(
    `Initialises correctly with a single media group on ${editor.name}`,
    { skip: ['edge'] },
    async (
      client: Parameters<typeof goToEditorTestingExample>[0],
      testName: string,
    ) => {
      const page = await goToEditorTestingExample(client);
      await mountEditor(page, {
        appearance: editor.appearance,
        defaultValue: JSON.stringify({
          version: 1,
          type: 'doc',
          content: [
            {
              type: 'mediaGroup',
              content: [
                {
                  type: 'media',
                  attrs: {
                    id: 'ced01e6d-e1a3-4b88-96c9-c5d042188e70',
                    type: 'file',
                    collection: 'jira-13085-field-description',
                    occurrenceKey: '8d6e3a67-1d21-4da7-9c2a-15610b79e40a',
                  },
                },
              ],
            },
          ],
        }),
        media: {
          allowMediaSingle: false,
          allowMediaGroup: true,
        },
      });

      const doc = await page.$eval(editable, getDocFromElement);
      expect(doc).toMatchCustomDocSnapshot(testName);
    },
  );
});
