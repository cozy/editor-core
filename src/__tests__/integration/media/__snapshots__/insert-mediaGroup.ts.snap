// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initialises correctly with a single media group on comment 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": null,
            "__fileName": "example.png",
            "__fileSize": null,
            "alt": null,
            "collection": "jira-13085-field-description",
            "height": null,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": "12345678-9abc-def0-1234-56789abcdef0",
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;

exports[`insert-mediaGroup.ts: Inserts a media group on comment 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "some text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "__contextId": "DUMMY-OBJECT-ID",
            "__displayType": null,
            "__external": false,
            "__fileMimeType": "image/svg+xml",
            "__fileName": "example.png",
            "__fileSize": 235,
            "alt": null,
            "collection": "MediaServicesSample",
            "height": null,
            "id": "1234-5678-abcd-efgh",
            "occurrenceKey": null,
            "type": "file",
            "url": null,
            "width": null,
          },
          "type": "media",
        },
      ],
      "type": "mediaGroup",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
